owncloudData/src/test/java/com/owncloud/android/data/storage/ScopedStorageProviderTest.kt

```diff
@@ -0,0 +1,330 @@
+
+        val isFolder = false
+        val parent = "$separator storage$separator emulated$separator 0$separator owncloud$separator".replace(" ", "")
+
+        val expectedPath = conditionsExpectedRemotePath(parent, newName, isFolder)
+        val actualPath = scopedStorageProvider.getExpectedRemotePath(remotePath, newName, isFolder)
+
+        assertEquals(expectedPath, actualPath)
+    }
+
+    @Test
+    fun `get expected remote path with parent variable without separator - ok - should return String expected remote path`() {
+
+        val isFolder = true
+        val parent = "$separator storage$separator emulated$separator 0$separator owncloud".replace(" ", "")
+
+        val expectedPath = conditionsExpectedRemotePath(parent, newName, isFolder)
+        val actualPath = scopedStorageProvider.getExpectedRemotePath(remotePath, newName, isFolder)
+
+        assertEquals(expectedPath, actualPath)
+    }
+
+    @Test
+    fun `get expected remote path with parent variable without separator and it is not a folder - ok - should return String expected remote path`() {
+        val isFolder = false
+        val parent = "$separator storage$separator emulated$separator 0$separator owncloud".replace(" ", "")
+
+        val expectedPath = conditionsExpectedRemotePath(parent, newName, isFolder)
+        val actualPath = scopedStorageProvider.getExpectedRemotePath(remotePath, newName, isFolder)
+
+        assertEquals(expectedPath, actualPath)
+    }
+
+    @Test(expected = IllegalArgumentException::class)
+    fun `test get expected remote path throws illegalArgumentException - ko - IllegalArgumentException`() {
+        val isFolder = false
+        val remotePath = ""
+
+        scopedStorageProvider.getExpectedRemotePath(remotePath, newName, isFolder)
+    }
+
+    @Test
+    fun `get temporal path if there are a Space - ok - String`() {
+        mockkStatic(Uri::class)
+        every { Uri.encode(accountName, "@") } returns uriEncoded
+
+        val temporalPathWithoutSpace = rootFolderPath + File.separator + "tmp" + File.separator + uriEncoded
+
+        val expectedValue = temporalPathWithoutSpace + File.separator + spaceId
+        val actualValue = scopedStorageProvider.getTemporalPath(accountName, spaceId)
+        assertEquals(expectedValue, actualValue)
+
+        verify (exactly = 1) {
+            scopedStorageProvider.getPrimaryStorageDirectory()
+        }
+    }
+
+    @Test
+    fun `get temporal path if there are not a Space - ok - String`() {
+        val spaceId = null
+
+        mockkStatic(Uri::class)
+        every { Uri.encode(accountName, "@") } returns uriEncoded
+
+        val expectedValue = rootFolderPath + File.separator + TEMPORAL_FOLDER_NAME + File.separator + uriEncoded
+        val actualValue = scopedStorageProvider.getTemporalPath(accountName, spaceId)
+        assertEquals(expectedValue, actualValue)
+    }
```

Comment:
Here we can add the same `verify` that you added in the other variant of this test

Reply:
Done